signer/src/ecdsa.rs

```diff
@@ -57,21 +69,24 @@ pub struct Signed<T> {
     pub signature: Vec<u8>,
 }
 
-impl<T: wsts::net::Signable> Signed<T> {
+impl<T> Signed<T>
+where
+    T: ProtoSerializable + Clone,
+    T: Into<<T as ProtoSerializable>::Message>,
+{
     /// Verify the signature over the inner data.
     pub fn verify(&self) -> bool {
-        self.inner
-            .verify(&self.signature, &self.signer_pub_key.into())
+        let msg = secp256k1::Message::from_digest(self.inner.digest());
+        let Ok(sig) = secp256k1::ecdsa::Signature::from_compact(&self.signature) else {
+            return false;
+        };
+
+        self.signer_pub_key.verify(SECP256K1, &msg, &sig).is_ok()
```

Comment:
I still need to finish going through this, but I guess we may have an issue with this approach: if I'm reading https://protobuf.dev/programming-guides/encoding/#implications correctly, we don't have any guarantee that `self.inner.digest()` will result in the same bytes as what the original sender signed; other than the specs, looking into the specific lib we use they don't explicitly guarantee that either.

Reply:
Right! I had forgotten about this. Okay, I'll revert a lot of these changes.